catalyst-toolbox/src/snapshot/influence_cap.rs

```diff
@@ -0,0 +1,200 @@
+use super::{Error, VoterEntry};
+use fraction::{BigFraction, Fraction};
+use rust_decimal::prelude::ToPrimitive;
+
+/// Calculates ceil(a / b) where a and b are integers with perfect precision
+#[inline]
+fn int_ceil(a: u64, b: u64) -> u64 {
+    assert!(b > 0);
+    (a + b - 1) / b
+}
+
+/// Let's denote with X the current voting power of user U and with TOT the total voting power
+/// and suppose X / TOT >= threshold.
+/// We want to remove a quantity Y from X such that (X - Y)/ (TOT - Y) <= treshold.
+/// The solution to this equation is therefore Y <= (X - T*TOT) / (1 - T).
+/// Since we work with integer numbers, we will use Y' = ceil(Y).
+///
+/// Panics if threshold is >= 1 or x/tot <= threshold
```

Comment:
I think `threshold` value could be equal to `1`.
Need to change `threshold is > 1`

Reply:
yep, the comment is outdated